src/main/java/io/deepstream/event/EventHandler.java

```diff
@@ -14,10 +17,18 @@
     private Emitter emitter;
     private Map options;
     private Connection connection;
+    private DeepstreamClient client;
+    private ResubscribeNotifier resubscribeNotifier;
 
-    public EventHandler( Map options, Connection connection ) {
+    public EventHandler( Map options, Connection connection, DeepstreamClient client ) {
         this.emitter = new Emitter();
         this.connection = connection;
+        this.client = client;
+        this.resubscribeNotifier = new ResubscribeNotifier(this.client, new ResubscribeCallback() {
```

Comment:
Resubscribe callback can be an interface which the event handler then implements and will make the callback:

 this.resubscribeNotifier = new ResubscribeNotifier( this.client, this )